### PR TITLE
fix(ext-plugin): when token is stale, refresh token and try again

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/ext-plugin/init.lua [hunk 1]
```diff
@@ -547,6 +533,34 @@ local function create_lrucache()
...
+    local ok, err, code, body
+    local tries = 0
+    while tries < 3 do
+        tries = tries + 1
+        ok, err, code, body = rpc_call(constants.RPC_HTTP_REQ_CALL, conf, ctx)
+        if ok then
+            if code then
+                return code, body
+            end
+
```
Comment (tzssangglass) [inline apisix/plugins/ext-plugin/init.lua:546]: why is there a blank line here?

File: apisix/plugins/ext-plugin/init.lua [hunk 2]
```diff
@@ -547,6 +533,34 @@ local function create_lrucache()
...
+            return
+        end
+
+        if not core.string.find(err, "conf token not found") then
```
Comment (tzssangglass) [inline apisix/plugins/ext-plugin/init.lua:550]: can we use code to determine this, I'm not sure

File: apisix/plugins/ext-plugin/init.lua (hunk 1 above)
Comment (spacewander) [reply apisix/plugins/ext-plugin/init.lua:546]: To make the `return` easy to find.

File: apisix/plugins/ext-plugin/init.lua (hunk 2 above)
Comment (spacewander) [reply apisix/plugins/ext-plugin/init.lua:550]: Yes

**Review by tokers — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Pull request merged**